pkgs/tools/package-management/nix/default.nix

```diff
@@ -119,9 +119,12 @@ common =
         [ "--with-store-dir=${storeDir}"
           "--localstatedir=${stateDir}"
           "--sysconfdir=${confDir}"
-          "--disable-init-state"
           "--enable-gc"
         ]
+        ++ lib.optionals (!is24) [
+          # option was removed in 2.4
+          "--disable-init-state"
+        ]
         ++ lib.optionals stdenv.isLinux [
```

Comment:
```suggestion
        ] ++ lib.optionals (!is24) [
          # option was removed in 2.4
          "--disable-init-state"
        ] ++ lib.optionals stdenv.isLinux [
```

Reply:
I'm trying to keep it consistent with the rest of the file